sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,189 @@
+import { DeepMocked, createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames, getISODateOnlyString } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../test/helpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  mockDownloadFiles,
+  saveFakeStudent,
+  saveFakeStudentRestriction,
+  RestrictionCode,
+} from "@sims/test-utils";
+import * as Client from "ssh2-sftp-client";
+import { Job } from "bull";
+import * as path from "path";
+import { SFASIntegrationScheduler } from "../sfas-integration.scheduler";
+import { Restriction, Student, StudentRestriction } from "@sims/sims-db";
+import { SystemUsersService } from "@sims/services";
+
+// SFAS received file mocks.
+const SFAS_LEGACY_RESTRICTION_FILENAME =
+  "SFAS-TO-SIMS-2024MAR07-LEGACY-RESTRICTIONS.txt";
+const SFAS_ALL_RESTRICTIONS_FILENAME =
+  "SFAS-TO-SIMS-2024MAR07-ALL-RESTRICTIONS.txt";
+
+describe(describeProcessorRootTest(QueueNames.SFASIntegration), () => {
+  let app: INestApplication;
+  let processor: SFASIntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let sfasDownloadFolder: string;
+  let studentExists: Student;
+  let sharedStudent: Student;
+  let systemUsersService: SystemUsersService;
+
+  beforeAll(async () => {
+    sfasDownloadFolder = path.join(__dirname, "sfas-files");
+    // Set the SFAS folder to the files mocks folders.
+    process.env.SFAS_RECEIVE_FOLDER = sfasDownloadFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(SFASIntegrationScheduler);
+    systemUsersService = nestApplication.get(SystemUsersService);
+    // Create student if it doesn't exist.
+    studentExists = await db.student.findOne({
```

Comment:
please use `sharedStudent` here and the other variable is not required.

Reply:
Keeping it as discussed.

Reply:
Thanks for making the change as discussed.